Library/Homebrew/cask/audit.rb

```diff
@@ -277,7 +277,7 @@ def audit_sha256_invalid
     sig { void }
     def audit_latest_with_livecheck
       return unless cask.version&.latest?
-      return unless cask.livecheckable?
+      return unless cask.livecheck_block?
```

Comment:
How about:
```suggestion
      return unless cask.livecheck?
```
Would that naming work?

Reply:
It makes sense to me, as `#livecheck` returns a [`Livecheck` object](https://github.com/Homebrew/brew/blob/84823d80f71a01eba4febbbdf4259a687219885b/Library/Homebrew/livecheck.rb) (the DSL values), so `#livecheck?` would align with that. As long as it's understandable to folks who aren't familiar with livecheck (but at least know it's an optional block in packages/resources), that works for me.

If we go that route, we would need to use a different name than `@livecheck` for the related instance variable, as that's already used for the `Livecheck` object. I'm using `has_livecheck_block` for this in some places, as I wasn't sure if `has_livecheck` is equally clear. If you think that folks will understand that `has_livecheck` means "has a `livecheck` block", then reducing the verbosity would be nice.

I'm currently using "livecheck block?:" as the label for the value in the `brew livecheck` debug output but we could use "Has livecheck?:" if we want to avoid "block" language everywhere. That would also align with a `has_livecheck` variable name (if used). For what it's worth, I don't think that using "livecheck?:" in that context would be as clear as the aforementioned options but that's just me.

Reply:
> It makes sense to me, as `#livecheck` returns a [`Livecheck` object](https://github.com/Homebrew/brew/blob/84823d80f71a01eba4febbbdf4259a687219885b/Library/Homebrew/livecheck.rb) (the DSL values), so `#livecheck?` would align with that.

Another option would be `bottle_defined?` which is the equivalent for the `livecheck` block.

Reply:
Looks like we also have `post_install_defined?` and `resource_defined?`, so it may be nice to follow the existing pattern. I can update this to `livecheck_defined?` if that makes sense to you.

If we do that, we can also use a similar `livecheck_defined` name in place of `has_livecheck_block`.

Reply:
Yup, sounds good thanks!